tensorflow/python/estimator/inputs/pandas_io.py

```diff
@@ -78,6 +96,9 @@ def pandas_input_fn(x,
     raise TypeError('shuffle must be explicitly set as boolean; '
                     'got {}'.format(shuffle))
 
+  if isinstance(target_column, list):
+    raise TypeError('target_column must be a string or None')
```

Comment:
I don't think None can work. So, this error message is not accurate. Can we just check whether target_column is six.string_types. 

For example, if user passes a dict, the error message, triggered by downstream, will be very confusing. It is better to just check against we need here. Thanks.

Reply:
Hi @xiejw, took that suggestion and am now throwing error if target_column is not an instance of six.string_types